server/store/store_test.go

```diff
@@ -150,3 +233,210 @@ func TestCheckEnabledTeamByTeamID(t *testing.T) {
+	defer tearDownContainer()
+
+	mockPostInfo := storemodels.PostInfo{
+		MattermostID:        "mockMattermostID",
+		MSTeamsID:           "mockMSTeamsID",
+		MSTeamsChannel:      "mockMSTeamsChannel",
+		MSTeamsLastUpdateAt: time.UnixMicro(int64(100)),
+	}
+
+	storeErr := store.LinkPosts(mockPostInfo)
+	assert.Nil(storeErr)
+
+	resp, getErr := store.GetPostInfoByMSTeamsID("mockMSTeamsChannel", "mockMSTeamsID")
+	assert.Equal(&mockPostInfo, resp)
+	assert.Nil(getErr)
+}
+
+func TestLinkPostsAndGetPostInfoByMattermostID(t *testing.T) {
+	assert := assert.New(t)
+	store, _, tearDownContainer := setupTestStore(&plugintest.API{}, model.DatabaseDriverPostgres)
+	defer tearDownContainer()
+
+	mockPostInfo := storemodels.PostInfo{
+		MattermostID:        "mockMattermostID",
+		MSTeamsID:           "mockMSTeamsID",
+		MSTeamsChannel:      "mockMSTeamsChannel",
+		MSTeamsLastUpdateAt: time.UnixMicro(int64(100)),
+	}
+
+	storeErr := store.LinkPosts(mockPostInfo)
+	assert.Nil(storeErr)
+
+	resp, getErr := store.GetPostInfoByMattermostID("mockMattermostID")
+	assert.Equal(&mockPostInfo, resp)
+	assert.Nil(getErr)
+}
+
+func TestSetUserInfoAndTeamsToMattermostUserID(t *testing.T) {
+	assert := assert.New(t)
+	store, _, tearDownContainer := setupTestStore(&plugintest.API{}, model.DatabaseDriverMysql)
+	defer tearDownContainer()
+	store.encryptionKey = func() []byte {
+		return make([]byte, 16)
+	}
+
+	storeErr := store.SetUserInfo(testutils.GetID(), testutils.GetTeamUserID(), &oauth2.Token{})
+	assert.Nil(storeErr)
+
+	resp, getErr := store.TeamsToMattermostUserID(testutils.GetTeamUserID())
+	assert.Equal(testutils.GetID(), resp)
+	assert.Nil(getErr)
+}
+
+func TestSetUserInfoAndMattermostToTeamsUserID(t *testing.T) {
+	assert := assert.New(t)
+	store, _, tearDownContainer := setupTestStore(&plugintest.API{}, model.DatabaseDriverPostgres)
+	defer tearDownContainer()
+	store.encryptionKey = func() []byte {
+		return make([]byte, 16)
+	}
+
+	storeErr := store.SetUserInfo(testutils.GetID(), testutils.GetTeamUserID(), &oauth2.Token{})
+	assert.Nil(storeErr)
+
+	resp, getErr := store.MattermostToTeamsUserID(testutils.GetID())
+	assert.Equal(testutils.GetTeamUserID(), resp)
+	assert.Nil(getErr)
+}
+
+func TestSetUserInfoAndGetTokenForMattermostUser(t *testing.T) {
+	assert := assert.New(t)
+	store, _, tearDownContainer := setupTestStore(&plugintest.API{}, model.DatabaseDriverPostgres)
+	defer tearDownContainer()
+	store.encryptionKey = func() []byte {
+		return make([]byte, 16)
+	}
+
+	token := &oauth2.Token{
+		AccessToken:  "mockAccessToken",
+		RefreshToken: "mockRefreshToken",
+	}
+
+	storeErr := store.SetUserInfo(testutils.GetID(), testutils.GetTeamUserID(), token)
+	assert.Nil(storeErr)
+
+	resp, getErr := store.GetTokenForMattermostUser(testutils.GetID())
+	assert.Equal(token, resp)
+	assert.Nil(getErr)
+}
+
+func TestSetUserInfoAndGetTokenForMSTeamsUser(t *testing.T) {
```

Comment:
Most of these test cases need to cover the non-happy path. I would expect to test what happens here, for example, when I try to get a not existing MSTeamsUser.